qaframework-bdd-tests/src/test/resources/features/refapp-2.x/15-RolesAndPrivileges/RolesAndPrivileges.feature

```diff
@@ -0,0 +1,66 @@
+Feature: Roles & Privileges Management
+
+  @selenium
+  @rolesAndPrivileges
+  Scenario: Adding the privilege
+    Given User clicks on configure metadata link from home page
+    When User clicks on manage privileges link on the configure metadata page
+    And User clicks the Add New Privilege button
+    And User fills the new privilege form
+    And User clicks the save button
+
+  @selenium
+  @rolesAndPrivileges
+  Scenario: Editing the privilege
+    Given User clicks on configure metadata link from home page
+    When User clicks on manage privileges link on the configure metadata page
+    And User clicks the Add New Privilege button
+    And User fills the new privilege form
+    And User clicks the save button
+    And User search for the created privilege
+    And User edits privilege
+
+  @selenium
+  @rolesAndPrivileges
+  Scenario: Deleting the privilege
+    Given User clicks on configure metadata link from home page
+    When User clicks on manage privileges link on the configure metadata page
+    And User clicks the Add New Privilege button
+    And User fills the new privilege form
+    And User clicks the save button
+    And User search for the created privilege
+    And User clicks delete privilege
+    Then System confirms delete
+
+  @selenium
+  @rolesAndPrivileges
+  Scenario: Adding a new Role
+    Given User clicks on System Administration Link from home page
+    When User clicks on Advanced Administration link from the System Administration Page
+    Then User clicks on manage roles link on the advanced administration page
+    And User clicks the Add New Role button on the manage roles page
+    And User fills the new role form
+    And User saves role
+
+  @selenium
+  @rolesAndPrivileges
+  Scenario: Editing Role
+    Given User clicks on System Administration Link from home page
+    When User clicks on Advanced Administration link from the System Administration Page
+    Then User clicks on manage roles link on the advanced administration page
+    And User clicks the Add New Role button on the manage roles page
+    And User fills the new role form
+    And User saves role
+    And User edits the role
+
+  @selenium
+  @rolesAndPrivileges
+  Scenario: Delete the role
+    Given User clicks on System Administration Link from home page
+    When User clicks on Advanced Administration link from the System Administration Page
+    Then User clicks on manage roles link on the advanced administration page
+    And User clicks the Add New Role button on the manage roles page
+    And User fills the new role form
+    And User saves role
+    And User selects the role to be deleted
+    And User deletes role
```

Comment:
Always use gherkin syntax then to be the last scenario  checkout here https://www.toolsqa.com/cucumber/gherkin-keywords/  and heree https://cucumber.io/docs/cucumber/step-definitions/

Reply:
thanks for the enlightenment

Reply:
Welcome.

Reply:
Could you throw light on why this line https://github.com/openmrs/openmrs-contrib-qaframework/blob/795cc1ecefbbf661fa6d1afaf495a83ad8e6991d/qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/RolesAndPrivilegesSteps.java#L139 is causing the workflow to fail. The text is asserted but  the test fails to finish/destroy